### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by webstrand** [eslint/eslint#19653]
### What rule do you want to change?

no-useless-escape

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/[0\-]/
```

### What does the rule currently do for this code?

It currently marks `\-` as a useless escape

### What will the rule do after it's changed?

It will not mark `\-` as a useless escape in regex character sets

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

`/[0\-]/` while technically a useless escape, omitting the escape is a common source of errors when additional characters are added to the character set accidentally creating a character range. For example `/[0-]/` extended into `/[0-9]/` produces a very different outcome than adding a single character `9` to the character set.

**Comment by mdjermanovic** [comment]
I don't think this rule should allow any useless escapes by default, but I'd be in favor of adding an option to ignore certain characters. 

A similar proposal (https://github.com/eslint/eslint/issues/12856) didn't get much attention from the team, but since it was a long time ago, I'm curious what @eslint/eslint-team thinks now.

**Comment by nzakas** [comment]
I don't think we should allow any useless escapes by default. I'm neutral on adding an option so happy to go along with consensus.

**Comment by snitin315** [comment]
I'd be in favor of an option to ignore specific characters. Marking this as accepted to add a new option to ignore specific characters, while keeping the default behavior as is.

@webstrand Would you like to submit a pull request for this change?

**Comment by snitin315** [comment]
It could be something like `ignoredCharacters` or `allowedCharacters`, which would be an array of characters.

**Comment by nzakas** [comment]
👍 to `allowedCharacters`
